hazelcast/src/main/java/com/hazelcast/map/IMap.java

```diff
@@ -1762,6 +1764,82 @@ void set(@Nonnull K key, @Nonnull V value,
              long ttl, @Nonnull TimeUnit ttlUnit,
              long maxIdle, @Nonnull TimeUnit maxIdleUnit);
 
+    /**
+     * Copies all of the mappings from the specified map to this map without loading
+     * non-existing elements from map store (which is more efficient than {@code putAll()}).
+     * Do not concurrently mutate the given map.
```

Comment:
`Do not concurrently mutate the given map.` -> what does this mean? Should I as a user stop mutating the map during this call?

Reply:
For both of the comments, they are actually from putAll / putAllAsync that I follow.

`IMap#putAll` inherits doc from `Map#putAll` which says `The behavior of this operation is undefined if the specified map is modified while the operation is in progress`
`IMap#putAllAsync` says `This version doesn't support batching. Don't mutate the given map until the future completes.`

Since we share the underlying implementation between putAll vs setAll, I would imagine their javadoc to be similar to each other. Do you think we need to change the wordings to make it clearer?

Reply:
Yes, I'd prefer removing the comment about not mutating the map from both methods since it's misleading and users may think they need to synchronise their entire application to perform a single `putAll`/`setAll` call. 
Instead, the guarantee is simple - the operation isn't atomic and concurrent operations may or may not be reflected in the results of the `putAll`/`setAll` call. That's it. The call will not block or hang, even if you call other operations on the map concurrently.

Reply:
Got you. Should I remove `Don't mutate the given map until the future completes.` for `putAllAsync` as well, so that we are consistent?

Same for `This version doesn't support batching.` in both `setAllAsync` and `putAllAsync`.

Reply:
I think I would leave the `This version doesn't support batching` intact, since `MapProxyImpl` would try to apply batching according to `hazelcast.map.put.all.batch.size` for putAll / setAll. That comment specifically says batching is not supported for async version.

Reply:
Yes, you can remove the comment for not mutating from both and you're right, you can leave the batching comment on both.